sys/riscv/riscv/pmap.c

```diff
@@ -359,7 +359,10 @@ pagezero(void *p)
 #define	pmap_l2_index(va)	(((va) >> L2_SHIFT) & Ln_ADDR_MASK)
 #define	pmap_l3_index(va)	(((va) >> L3_SHIFT) & Ln_ADDR_MASK)
 
-#define	PTE_TO_PHYS(pte)	((pte >> PTE_PPN0_S) * PAGE_SIZE)
+#define	PTE_TO_PHYS(pte)	((((pte) & ~0xFFC0000000000000ULL) \
+					>> PTE_PPN0_S) * PAGE_SIZE)
+#define	PTE2_TO_PHYS(l2)	((((l2) & ~0xFFC0000000000000ULL) \
+					>> PTE_PPN1_S) << L2_SHIFT)
```

Comment:
This could easily be confused with getting the physical address of a non-leaf L2 entry, rather than being an L2 PTE -> superpage PA macro.

Reply:
`SUPERPTE_TO_PHYS`?